tests/foreman/virtwho/test_cli_virtwho.py

```diff
@@ -14,82 +14,327 @@
+        cls.hypervisor_password = settings.virtwho.hypervisor_password
+        cls.hypervisor_config_file = settings.virtwho.hypervisor_config_file
+        cls.vdc_physical = settings.virtwho.sku_vdc_physical
+        cls.vdc_virtual = settings.virtwho.sku_vdc_virtual
+
+    def _make_virtwho_configure(self):
+        args = {
+            u'debug': 1,
+            u'interval': '60',
+            u'hypervisor-id': 'hostname',
+            u'hypervisor-type': self.hypervisor_type,
+            u'hypervisor-server': self.hypervisor_server,
+            u'organization-id': 1,
+            u'filtering-mode': 'none',
+            u'satellite-url': self.satellite_url,
+        }
+        if self.hypervisor_type == 'libvirt':
+            args[u'hypervisor-username'] = self.hypervisor_username
+        elif self.hypervisor_type == 'kubevirt':
+            args[u'kubeconfig'] = self.hypervisor_config_file
+        else:
+            args['hypervisor-username'] = self.hypervisor_username
+            args['hypervisor-password'] = self.hypervisor_password
+        return args
+
+    @tier1
+    def test_positive_deploy_configure_by_id(self):
+        """ Verify " hammer virt-who-config deploy"
+
+        :id: 19ffe76e-7e3d-48c7-b846-10a83afe0f3e
+
+        :expectedresults: Config can be created and deployed
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
+        """
+        name = gen_string('alpha')
+        args = self._make_virtwho_configure()
+        args.update({'name': name})
+        vhd = VirtWhoConfig.create(args)['general-information']
+        self.assertEquals(vhd['status'], 'No Report Yet')
+        command = get_configure_command(vhd['id'])
+        hypervisor_name, guest_name = deploy_configure_by_command(command, debug=True)
+        self.assertEquals(
+            VirtWhoConfig.info({'id': vhd['id']})['general-information']['status'], 'OK')
+        hosts = [
+            (hypervisor_name, 'product_id={}'.format(self.vdc_physical)),
+            (guest_name, 'type=STACK_DERIVED')]
+        for hostname, sku in hosts:
+            host = Host.list({'search': hostname})[0]
+            vdc = Subscription.list({
+                'organization': DEFAULT_ORG,
+                'search': sku,
+            })[0]
+            result = Host.subscription_attach({
+                'host-id': host['id'],
+                'subscription-id': vdc['id']
+            })
+            self.assertTrue('attached to the host successfully' in '\n'.join(result))
+        VirtWhoConfig.delete({'name': name})
+        self.assertFalse(VirtWhoConfig.exists(search=('name', name)))
+
+    @tier1
+    def test_positive_deploy_configure_by_script(self):
+        """ Verify " hammer virt-who-config fetch"
+
+        :id: ef0f1e33-7084-4d0e-95f1-d3080dfbb4cc
+
+        :expectedresults: Config can be created, fetch and deploy
+
+        :CaseLevel: Integration
+
+        :CaseImportance: High
```

Comment:
as per test it looks `CaseImportance`: `Critical` , What do you think ?

Reply:
As above reply.